### PR TITLE
Add Contains Vector2 method to Box2Rotated

### PR DESCRIPTION
**Pull request by metalgearsloth**

### CODE REVIEW

File: Robust.Shared.Maths/Box2Rotated.cs [hunk 1]
```diff
@@ -149,6 +149,26 @@ public Box2Rotated(Box2 box, Angle rotation, Vector2 origin)
...
+        /// Convert a point in world-space coordinates to our local coordinates.
+        /// </summary>
+        private Vector2 GetLocalPoint(Vector2 point)
+        {
+            // Could make this more efficient but works for now I guess...
+            var boxCenter = Box.Center;
+
+            var result = point - boxCenter;
+            result = Origin + Rotation.RotateVec(result - Origin);
+            return result + boxCenter;
```
Comment (PaulRitter) [inline Robust.Shared.Maths/Box2Rotated.cs:169]: for the record, it took me 10 minutes and an a4 sheet of paper to understand this

Comment (PaulRitter) [reply Robust.Shared.Maths/Box2Rotated.cs:169]: im not sure i comprehend it fully but i trust the tests

**Review by PaulRitter — APPROVED** [review]
grafik

File: Robust.Shared.Maths/Box2Rotated.cs (hunk 1 above)
Comment (metalgearsloth) [reply Robust.Shared.Maths/Box2Rotated.cs:169]: Same dude

**Pull request merged**